staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -182,6 +190,120 @@ type SharedInformer interface {
 	SetWatchErrorHandler(handler WatchErrorHandler) error
 }
 
+// DoneChannel is a type alias for a channel that closes when an informer is stopped.
+type DoneChannel <-chan struct{}
+
+// OnListErrorFunc is a type alias for function that takes an error
+// and determines whether or not to stop.
+type OnListErrorFunc func(error) bool
+
+// StopHandle gives the caller of a shared informer information
+// on when and why the informer stopped.
+type StopHandle interface {
+	// Done indicates when the informer has been stopped.
+	Done() DoneChannel
+	// Err returns the error (if applicable) that caused the shared informer to be stopped.
+	// A nil error after h.Done is closed means that the informer was stopped externally.
+	// Err can only be called after h.Done() has closed otherwise it will return
+	// nil.
+	Err() error
+	// Close allows for stopping the ifnormer (and closing the done channel).
+	Close()
+	// WithError provides a way to set the error on the stop handle.
+	WithError(err error)
+	// MergeChan takes a chan and merges with the done channel such that the done channel closes
+	// when either the channel being merged closes or an explicit Close() is called on the stop handle.
+	MergeChan(ch <-chan struct{}) context.CancelFunc
+}
+
+// stopHandle implements the StopHandle interface.
+type stopHandle struct {
+	// stopWrite is used to close the stopRead channel when the informer's
+	// underlying controller/reflector stop running.
+	stopWrite chan<- struct{}
+	// stopRead is closed in order to stop the informer.
+	stopRead <-chan struct{}
+	// err gets populated by the cause of the informer stopping
+	// or is nil because the stopRead channel was closed.
+	err error
+}
+
+// NewStopHandle creates a stop handle from a bidirectonal stop channel.
+func NewStopHandle(stop chan struct{}) StopHandle {
+	return &stopHandle{
+		stopWrite: stop,
+		stopRead:  stop,
+	}
+}
+
+// Done returns a channel that's closed when the informer is stopped.
+func (h *stopHandle) Done() DoneChannel {
+	return h.stopRead
+}
+
+// Err returns the error (if applicable) that caused the shared informer to be stopped.
+// A nil error after h.Done is closed means that the informer was stopped externally.
+// Err can only be called after h.Done() has closed otherwise it will return
+// nil.
+func (h *stopHandle) Err() error {
+	return h.err
```

Comment:
what about adding the check for if h.Done() has closed?

Reply:
custom stopHandle has been removed